dev_scripts/release-container-image.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+"""
+Script to automate the release of a new Dangerzone container image.
+
+This script automates the manual steps described in the container image release
+process, including:
+1. Picking a release candidate image
+2. Attesting provenance and reproducibility
+3. Signing and publishing the image
+"""
+
+import json
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+import click
+
+PROJECT_ROOT = Path(__file__).parent.parent
+IMAGE_NAME_PATH = PROJECT_ROOT / "share" / "image-name.txt"
+DOCKER_ENV_PATH = PROJECT_ROOT / "Dockerfile.env"
+
+
+def extract_debian_archive_date():
+    with DOCKER_ENV_PATH.open() as f:
+        content = f.read()
+
+    match = re.search(r"DEBIAN_ARCHIVE_DATE=(\d+)", content)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+
+def run(cmd, check=True, capture_output=False):
+    """Run a command and optionally capture output."""
+    click.echo(f"Running: {' '.join(cmd)}")
+    result = subprocess.run(cmd, check=check, capture_output=capture_output, text=True)
+    return result
+
+
+@click.command()
+@click.option("--commit", default=None)
+@click.option(
+    "--ghcr-signer-path",
+    type=click.Path(exists=True),
+    required=True,
+    help="Path to the ghcr-signer repository",
+    default="../ghcr-signer",
+)
+@click.option("--repository", default="freedomofpress/dangerzone")
+@click.option("--branch", default="main")
+def attest_release_sign(commit, ghcr_signer_path, repository, branch):
+    """
+    Attest the provenance of a commit, reproduce its digest and prepare signatures
+
+    Usage: release-container-image
+    """
+
+    # Get commit hash
+    if not commit:
+        result = run(
+            ["git", "rev-parse", "HEAD"],
+            capture_output=True,
+        )
+        commit = result.stdout.strip()
+
+    full_image = get_candidate_image(commit)
+    attest_provenance(full_image, commit, repository, branch)
+    digests = get_platform_digests(full_image)
+    debian_archive_date = extract_debian_archive_date()
+    for platform in ["linux/amd64", "linux/arm64"]:
+        pass  # reproduce_image(full_image, platform, debian_archive_date, digests[platform])
```

Comment:
Reminder to uncomment this before merging. Also, kind of an important thing: this step takes quite a while in most cases. We should probably have a flag that can bypass this step, just in case we have performed it already, provided that this flag's name is very strict (e.g, `--yes-i-have-reproduced <digest1>,<digest2>`)